src/main/webapp/app/overview/discussion/discussion.component.ts

```diff
@@ -0,0 +1,125 @@
+    ngOnChanges(): void {
+        this.resetCourseAndPosts();
+    }
+
+    /**
+     * @private sets the course that is associated with the exercise or lecture given as input, triggers method on metis service to fetch (an push back via subscription)
+     * exercise or lecture posts respectively, creates an empty default post
+     */
+    private resetCourseAndPosts(): void {
+        this.course = this.exercise ? this.exercise.course! : this.lecture!.course!;
+        this.courseId = this.course.id!;
+        this.metisService.setCourse(this.course);
+        this.metisService.getPostsForFilter({ exercise: this.exercise, lecture: this.lecture });
+        this.createdPost = this.createEmptyPost();
+    }
+
+    /**
+     * makes discussion expandable by configuring 'interact'
+     */
+    ngAfterViewInit(): void {
+        interact('.expanded-discussion')
+            .resizable({
+                edges: { left: '.draggable-left', right: false, bottom: false, top: false },
+                modifiers: [
+                    // Set maximum width
+                    interact.modifiers!.restrictSize({
+                        min: { width: 300, height: 0 },
+                        max: { width: 600, height: 4000 },
+                    }),
+                ],
+                inertia: true,
+            })
+            .on('resizestart', function (event: any) {
+                event.target.classList.add('card-resizable');
+            })
+            .on('resizeend', function (event: any) {
+                event.target.classList.remove('card-resizable');
+            })
+            .on('resizemove', function (event: any) {
+                const target = event.target;
+                target.style.width = event.rect.width + 'px';
+            });
+    }
+
+    /**
+     * creates empty default post that is needed on initialization of a newly opened modal to edit or create a post
+     * @return Post created empty default post
+     */
+    createEmptyPost(): Post {
+        const post = new Post();
+        post.content = '';
+        post.visibleForStudents = true;
+        if (this.exercise) {
+            post.exercise = {
+                ...this.exerciseService.convertExerciseForServer(this.exercise),
+            };
+        } else {
+            post.lecture = {
+                id: this.lecture!.id,
+            };
+        }
+        return post;
+    }
+
+    /**
+     * defines a function that returns the post id as unique identifier,
+     * by this means, Angular determines which post in the collection of posts has to be reloaded/destroyed on changes
+     */
+    postsTrackByFn(index: number, post: Post): number {
+        return post.id!;
+    }
+
+    /**
+     * resets createdPost to a new default post after a post was successfully created
+     */
+    onCreatePost(): void {
+        this.createdPost = this.createEmptyPost();
+    }
+
+    ngOnDestroy(): void {
+        this.postsSubscription.unsubscribe();
+    }
```

Comment:
I think it is particularly important, if the observable does emit more than once as it is the case here (Behavior Subject), but not for Observables returned by the HttpClient (they should only emit once). In fact, I think it is not necessary here, as the MetisService (which returns the posts observable) is provided only for this `discussion.component` (and will thus be destroyed anyways when this component gets destroyed). But I wanted to be explicit here, as this might change in the future and then we have starving subscriptions all over the place.